lib/OpenQA/Script.pm

```diff
@@ -45,9 +45,10 @@ sub clone_job_apply_settings {
     $settings->{is_clone_job} = 1;    # used to figure out if this is a clone operation
 
     for my $arg (@$argv) {
+        next unless $arg;
```

Comment:
Not sure why empty arguments should be completely skipped.
But at least it should check for `length`. Otherwise an argument like `0` will be skipped too.

Reply:
yeah but we only allow `$key=$value` anyway. This line is just *allowing* empty arguments but silently ignoring them. The next `unless` would make sure they are skipped anyway but then with a warning printed which I would prevent here.

Reply:
But silently skipping `0`  is not helpful I think.

Reply:
Fine. I reverted this one line and rather fix the place where we call it with an empty argument in openqa-clone-custom-git-refspec